### ISSUE TITLE
[2020] Towards a Human-like Open-Domain Chatbot

### ISSUE THREAD

**Issue by cfiken** [cfiken/paper-reading#39]
### タイトル
Towards a Human-like Open-Domain Chatbot
### リンク
https://arxiv.org/abs/2001.09977
### 著者
Daniel Adiwardana, Minh-Thang Luong, David R. So, Jamie Hall, Noah Fiedel, Romal Thoppilan, Zi Yang, Apoorv Kulshreshtha, Gaurav Nemade, Yifeng Lu, Quoc V. Le
### 投稿日付
2020/01/27
## 1. どんなもの？

人間による会話評価の分析を行い、人間評価用の指標である SSA (Sensibleness and Specificity Average) を提案し、これがモデルの perplexity と相関があることを実験により確認した。
perplexity について最適化を行ったモデルとして、マルチターンの雑談 chatbot である Meena を提案。提案指標である SSA で既存の雑談モデルを大きく上回るスコア(72%)となった。

## 2. 先行研究と比べてどこがすごい？

大規模データ(約341GBのテキスト)・大規模モデル(2.6B のパラメータを持つ Evolved Transformer)で、人間による会話評価の分析に基づいて行った学習モデルで既存スコアを大きく更新した。
perplexity の低いモデルが良い人間評価に相関することを実験により示した。

## 3. 技術や手法のキモはどこ？

- 人間評価の分析から SSA の提案、及び相関する自動評価指標として perplexity の選択。 
- Enc 1層、Dec 13層の Evolved Transformer を使用して、超大規模学習 (後述) を行ったモデル。
- decoding を工夫。既存で主要に使われている beam search ではなく、シンプルな sampling-and-rerank の手法で行う。
    - softmax 計算時に temperature T を加えて分布を調整し、N件の候補をサンプリング後、スコアなどで re-rank を行う
- Meena (full) として更に decoding を改善。サンプリングに top-k を使用し、temperature を調整した上で、同じような分を繰り返す(cross-turn repetition) を軽減するためにルールで候補を削除。

## 4. どうやって有効だと検証した？

- 既存 chatbot モデルと一緒に人間による評価を行い、Meena が最も高い SSA を獲得。人間スコアである 86% に対して Meena (base) が 72%, Meena (full) が 79% と、かなり人間に近いスコアとなった。

## 5. 議論はある？

- 妥当なテストになっているかどうか。static test と interactive test を行ったが、どちらもまだ課題がある。
    - static test は context が 1-3 と小さい
    - interactive test は他モデルとの比較がしづらく、また14-28ターンはまだ小さい (挨拶が一部を占めるので、深いトピックについては話せない)
- 指標として Sensibleness と Specificity にまとめたが、これがベストというわけではない
    - 他にも多くの指標が考えられる (humor, empathy, ...)
    - Sensibleness を分解してもっと細かく見ることもできる (logical, personal consistency, ...)

## 6. 次に読むべき論文は？

- Evolved Transformer
- Top K など decoding strategy 周りの論文

**Comment by cfiken** [comment]
# 7. 実験詳細

- モデルは 2.6B のパラメータを持つ Evolved Transformer
    - 1個の Encoder と 13個の Decoder
    - hidden units は 2,560個, attention head は 32, 1文の length は 128
    - (GPT2 でも 1.5B)
- データは social media などから集めて filtering した 341GB のデータセット
    - (context, response) のペアで 867M 個
    - word にして 40B 個, BPE にして 61B 個
- 学習
    - TPU-v3 Pod (2,048 cores) で30日間 (...!!)
    - overfit したので dropout 0.1 入れた
    - メモリ節約のため Adafactor optimizer
